api/types/swarm/task.go

```diff
@@ -66,7 +66,8 @@ type TaskSpec struct {
 	// parameters have been changed.
 	ForceUpdate uint64
 
-	Runtime RuntimeType `json:",omitempty"`
+	Runtime    RuntimeType `json:",omitempty"`
+	PluginSpec *PluginSpec `json:",omitempty"`
```

Comment:
I'd suggest putting this next to `ContainerSpec` in the struct definition and adding a comment that they are mutually exclusive. We should probably change `ContainerSpec` to be a pointer (and make sure no code assumes it exists).